contracts/l2/dai.sol

```diff
@@ -82,53 +81,87 @@ contract Dai {
   }
 
   // --- ERC20 Mutations ---
-  function transfer(address dst, uint256 wad) external returns (bool) {
-    return transferFrom(msg.sender, dst, wad);
+  function transfer(address to, uint256 value) external returns (bool) {
+    uint256 balance = balanceOf[msg.sender];
+    require(balance >= value, "Dai/insufficient-balance");
+
+    balanceOf[msg.sender] = balance - value;
+    balanceOf[to] = add(balanceOf[to], value);
+
+    emit Transfer(msg.sender, to, value);
+
+    return true;
   }
-  function transferFrom(address src, address dst, uint256 wad) public returns (bool) {
-    require(balanceOf[src] >= wad, "Dai/insufficient-balance");
+  function transferFrom(address from, address to, uint256 value) external returns (bool) {
+    uint256 balance = balanceOf[from];
+    require(balance >= value, "Dai/insufficient-balance");
 
-    if (src != msg.sender && allowance[src][msg.sender] != type(uint256).max) {
-        require(allowance[src][msg.sender] >= wad, "Dai/insufficient-allowance");
+    if (from != msg.sender) {
+      uint256 allowed = allowance[from][msg.sender];
+      if (allowed != type(uint256).max) {
+        require(allowed >= value, "Dai/insufficient-allowance");
 
-        allowance[src][msg.sender] = sub(allowance[src][msg.sender], wad);
+        allowance[from][msg.sender] = allowed - value;
+      }
     }
 
-    balanceOf[src] = sub(balanceOf[src], wad);
-    balanceOf[dst] = add(balanceOf[dst], wad);
+    balanceOf[from] = balance - value;
+    balanceOf[to] = add(balanceOf[to], value);
+
+    emit Transfer(from, to, value);
+
+    return true;
+  }
+  function approve(address spender, uint256 value) external returns (bool) {
+    allowance[msg.sender][spender] = value;
+
+    emit Approval(msg.sender, spender, value);
+
+    return true;
+  }
+  function increaseAllowance(address spender, uint256 addedValue) external returns (bool) {
+    uint256 newValue = add(allowance[msg.sender][spender], addedValue);
+    allowance[msg.sender][spender] = newValue;
 
-    emit Transfer(src, dst, wad);
+    emit Approval(msg.sender, spender, newValue);
 
     return true;
   }
-  function approve(address usr, uint256 wad) external returns (bool) {
-    allowance[msg.sender][usr] = wad;
+  function decreaseAllowance(address spender, uint256 subtractedValue) external returns (bool) {
+    uint256 allowed = allowance[msg.sender][spender];
+    require(allowed >= subtractedValue, "Dai/insufficient-allowance");
```

Comment:
Would be nice if this function were more consistent w/`increaseAllowance`. Personally, I don't think there's a strong case to give a revert message here but not in `increaseAllowance`, so I'd move both to the more succinct format (and use consistent local var naming, probably "newValue" in both cases).

Reply:
In the current prod version of Dai there is revert messages on all subtractions, but not on additions. This makes sense to me as underflow will be a lot more common for the average user. I opted to maintain this pattern which is why you see the difference here between `increaseAllowance` and `decreaseAllowance`.